app/src/main/java/com/example/habithelper/activities/AccountCreationActivity.java

```diff
@@ -39,113 +37,138 @@ public class SetupAccount extends AppCompatActivity {
             @Override
             public void onClick(View v) {
-                name = etName.getText().toString();
-                zipString = etZip.getText().toString();
-                customHabitOne=etCustomOne.getText().toString();
-                customHabitTwo=etCustomTwo.getText().toString();
-
-                if(checkBoxCustomOne.isChecked()){
-                    if (customHabitOne.isEmpty()) {
-                        Toast.makeText(SetupAccount.this, "habit cannot be empty", Toast.LENGTH_SHORT).show();
-                        return;
-                    }
-                }
+                // takes in the user input and ensures it meets requirements (e.g. num habits selected)
+                checkUserInput();
 
-                if(checkBoxCustomTwo.isChecked()){
-                    if (customHabitTwo.isEmpty()) {
-                        Toast.makeText(SetupAccount.this, "habit cannot be empty", Toast.LENGTH_SHORT).show();
-                        return;
-                    }
-                }
+                // checking whether or not the zipcode inputted is valid
+                checkZipCode();
+            }
+        });
+    }
 
-                if (name.isEmpty()) {
-                    Toast.makeText(SetupAccount.this, "name cannot be empty", Toast.LENGTH_SHORT).show();
-                    return;
-                }
-                if (zipString.isEmpty()) {
-                    Toast.makeText(SetupAccount.this, "zipcode cannot be empty", Toast.LENGTH_SHORT).show();
-                    return;
-                }
+    /**
+     * Checks whether or not the zipcode the user inputted is a valid zipcode
+     * If it is not valid, tryToInitialize will be called with validZip = false and will not execute.
+     */
+    private void checkZipCode() {
+        AsyncHttpClient client = new AsyncHttpClient();
+        String api_request = GET_WEATHER_URL + zipString;
+        client.get(api_request, new JsonHttpResponseHandler() {
 
-                // initializing the list of habits the user wishes to track
-                initializeHabitList();
-                if (habitsList.size()<4) {
-                    Toast.makeText(SetupAccount.this, "Add more habits", Toast.LENGTH_SHORT).show();
-                    habitsList.clear();
-                    return;
-                }
+            @Override
+            public void onSuccess(int statusCode, Headers headers, JSON json) {
+                validZip = true;
+                tryToInitialize(validZip);
+            }
 
-                if (habitsList.size()>10) {
-                    Toast.makeText(SetupAccount.this, "Select fewer habits", Toast.LENGTH_SHORT).show();
-                    habitsList.clear();
-                    return;
-                }
+            @Override
+            public void onFailure(int statusCode, Headers headers, String response, Throwable throwable) {
+                validZip = false;
+                tryToInitialize(validZip);
```

Comment:
Since the variable validZip seems to not be used anywhere else, You can directly call tryToInitialize(false). Same for the True case above.